### DISCUSSION TITLE
Communicating between Bonsai and an external Python script

### DISCUSSION THREAD

**Discussion by vchundru2002** [bonsai-rx/bonsai#1708] [Q&A] [answered]
Hi, 

I would like to apply a model to the data stream in real time using a Python library that is not available in C# or IronPython. Is it possible to run Python scripts within a Bonsai node? If I were to set up a communication between the Python script so that the Bonsai workflow writes some features across various time points to a file, Python loads the file and applies a model, Python writes the classification results to another file, and Bonsai reads the new results. Is this at all possible? Are there any potential problems with analyzing data in this way?

**Comment by glopesdev** [comment]
@vchundru2002 Probably the most effective way to do this would be to use the new Python scripting package:
https://bonsai-rx.org/python-scripting/

**Reply by vchundru2002** [reply]
I tried using this, but I get this error: 
![image](https://github.com/bonsai-rx/bonsai/assets/60073624/153579b2-0bff-44cb-94f1-cc1230b70b35)

I tried going to the listed link and changing my PYTHONNET_PYDLL variable to the path of my python310.dll file in my conda environment, but the error is still there. Do you know what the issue could be?

**Reply by glopesdev** [reply, marked as answer]
@vchundru2002 conda environments are not directly supported by the package. I would recommend trying first with a standard python distribution and a virtual environment, as described in the installation instructions linked above. Once you get the basic setup going it may be easier to try variations.
